contracts/payment/LemonadeEscrowV1.sol

```diff
@@ -0,0 +1,208 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/finance/PaymentSplitter.sol";
+
+import "./DelegateManager.sol";
+import "./ILemonadeEscrow.sol";
+
+contract LemonadeEscrowV1 is
+    ILemonadeEscrow,
+    Ownable,
+    PaymentSplitter,
+    DelegateManager
+{
+    uint64 internal _startTime;
+    uint64 internal _endTime;
+
+    bool internal _closed;
+    uint16 internal _hostRefundPercent;
+    RefundPolicy[] internal _refundPolicies;
+
+    mapping(uint256 => bool) _paymentCancelled; //-- map paymentId
+    mapping(address => mapping(uint256 => Deposit[])) _deposits; //-- map user -> paymentId
+
+    constructor(
+        address owner,
+        address[] memory delegates,
+        address[] memory payees,
+        uint256[] memory shares,
+        uint64 startTime,
+        uint64 endTime,
+        uint16 hostRefundPercent,
+        RefundPolicy[] memory refundPolicies
+    ) PaymentSplitter(payees, shares) DelegateManager(delegates) {
+        require(startTime < endTime, "startTime must be smaller than endTime");
+        require(hostRefundPercent <= 100, "Invalid hostRefundPercent");
+
+        //-- check valid refundPolicies
+        if (refundPolicies.length == 1) {
+            _assertValidRefundPercent(refundPolicies[0]);
+        } else if (refundPolicies.length > 1) {
+            _assertValidRefundPercent(refundPolicies[0]);
+
+            for (uint256 i = 0; i < refundPolicies.length - 1; i++) {
+                require(
+                    refundPolicies[i].timestamp < refundPolicies[i + 1].timestamp 
+                    && refundPolicies[i].percent > refundPolicies[i + 1].percent,
+                    "Invalid refund policy order & percent"
+                );
+
+                _assertValidRefundPercent(refundPolicies[i + 1]);
+            }
+        }
```

Comment:
```suggestion
       for (uint256 i = 0; i < refundPolicies.length - 1; i++) {
                _assertValidRefundPercent(refundPolicies[i]);
                require(
                    refundPolicies[i].timestamp < refundPolicies[i + 1].timestamp 
                    && refundPolicies[i].percent > refundPolicies[i + 1].percent,
                    "Invalid refund policy order & percent"
                );
            }
```

Wonder if we can just do this? Besides, wouldn't you potentially get index out of bound if you do `i+1` while looping to the the end? 🤔

Reply:
this loop is to check whether the order of the policies is in desired format: the percentage is decrement and the timestamp is increment, also the percent need to be <= 100%

so there is 3 case:
- if the refund policies is empty then do nothing
- if the refund policies has just one element: just check the percent
- if there is more than 1 element: note that the for loop is from uint256 i = 0; i < **refundPolicies.length - 1**; i++ which mean this loop is to compare policy[i] & policy[i+1] against the above rule. This will never out of bound because i + 1 is always < refundPolicies.length.